e2e/metrics_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !cluster_proxy
+
+package e2e
+
+import (
+	"testing"
+
+	"github.com/coreos/etcd/pkg/testutil"
+)
+
+func TestV3MetricsSecure(t *testing.T)   { testV3Metrics(t, false) }
+func TestV3MetricsInsecure(t *testing.T) { testV3Metrics(t, true) }
+func testV3Metrics(t *testing.T, insecureMetricsURL bool) {
+	defer testutil.AfterTest(t)
+
+	cfg := configTLS
+	cfg.metricsURL = true
```

Comment:
possible to simplify the config by replacing the `metricsURL` and `insecureMetricsURL` with `metricsURLScheme string` with empty string meaning no metrics url:
```go
func TestV3MetricsSecure(t *testing.T)   { testV3Metrics(t, "https") }
func testV3Metrics(t *testing.T, scheme string) {
	defer testutil.AfterTest(t)

	cfg := configTLS
	cfg.metricsURLScheme = scheme
        ...
}
```